bundles/persistence/org.openhab.persistence.dynamodb/java/org/openhab/persistence/dynamodb/internal/DynamoDBPersistenceService.java

```diff
@@ -0,0 +1,444 @@
+import org.openhab.core.persistence.QueryablePersistenceService;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapperConfig;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapperConfig.PaginationLoadingStrategy;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.amazonaws.services.dynamodbv2.datamodeling.PaginatedQueryList;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.ComparisonOperator;
+import com.amazonaws.services.dynamodbv2.model.Condition;
+import com.amazonaws.services.dynamodbv2.model.CreateTableRequest;
+import com.amazonaws.services.dynamodbv2.model.GlobalSecondaryIndex;
+import com.amazonaws.services.dynamodbv2.model.ProvisionedThroughput;
+import com.amazonaws.services.dynamodbv2.model.ResourceNotFoundException;
+import com.amazonaws.services.dynamodbv2.model.TableDescription;
+import com.amazonaws.services.dynamodbv2.model.TableStatus;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * This is the implementation of the DynamoDB {@link PersistenceService}. It persists item values
+ * using the <a href="https://aws.amazon.com/dynamodb/">Amazon DynamoDB</a> time series database. The states (
+ * {@link State}) of an {@link Item} are persisted in a time series with names equal to the name of
+ * the item. All values are stored using integers or doubles, {@link OnOffType} and
+ * {@link OpenClosedType} are stored using 0 or 1.
+ *
+ * The default database name is "openhab"
+ *
+ */
+public class DynamoDBPersistenceService implements QueryablePersistenceService {
+
+    private ItemRegistry itemRegistry;
+    private DynamoDBClient db;
+    private static final Logger logger = LoggerFactory.getLogger(DynamoDBPersistenceService.class);
+    private boolean isProperlyConfigured;
+    private DynamoDBConfig dbConfig;
+    private DynamoDBTableNameResolver tableNameResolver;
+
+    /**
+     * For testing
+     *
+     * @return
+     */
+    DynamoDBClient getDb() {
+        return db;
+    }
+
+    public void setItemRegistry(ItemRegistry itemRegistry) {
+        this.itemRegistry = itemRegistry;
+    }
+
+    public void unsetItemRegistry(ItemRegistry itemRegistry) {
+        this.itemRegistry = null;
+    }
+
+    public void activate(final BundleContext bundleContext, final Map<String, Object> config) {
+        logger.debug("dynamodb persistence service activated");
+        resetClient();
+        try {
+            dbConfig = DynamoDBConfig.fromConfig(config);
+            if (dbConfig == null) {
+                throw new IllegalArgumentException("Something was wrong with configuration");
```

Comment:
If you look the implementation of the static method fromConfig, you can find out the cases when it returns null. These include missing AWS region (this is error), and missing configuration parameters. 

So I do think this is an error with configuration.

But what does it mean if `config`  is `null`? This binding has mandatory parameters and cannot work without configuration. How should I handle that?